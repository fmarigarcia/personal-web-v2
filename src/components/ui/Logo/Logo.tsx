import React from 'react';
import clsx from 'clsx';

interface LogoProps {
    onClick?: () => void;
    className?: string;
}

export const Logo: React.FC<LogoProps> = ({ onClick, className = '' }) => {
    return (
        <div className={clsx('flex items-center', className)}>
            <button
                onClick={onClick}
                className="text-xl font-bold text-stone-800 hover:text-stone-950 transition-colors"
                type="button"
                aria-label="Go to home"
            >
                FM
            </button>
        </div>
    );
};

export default Logo;
