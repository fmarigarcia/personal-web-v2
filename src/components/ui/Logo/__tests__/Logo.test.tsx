import { render, screen, fireEvent } from '@testing-library/react';
import { Logo } from '../Logo';

describe('Logo', () => {
    it('should render logo with default text', () => {
        render(<Logo />);

        const logo = screen.getByRole('button', { name: /go to home/i });
        expect(logo).toBeInTheDocument();
        expect(logo).toHaveTextContent('FM');
    });

    it('should handle click when onClick prop is provided', () => {
        const mockOnClick = jest.fn();
        render(<Logo onClick={mockOnClick} />);

        const logo = screen.getByRole('button');
        fireEvent.click(logo);

        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });

    it('should not crash when clicked without onClick prop', () => {
        render(<Logo />);

        const logo = screen.getByRole('button');
        expect(() => fireEvent.click(logo)).not.toThrow();
    });

    it('should apply custom className', () => {
        const customClass = 'custom-logo';
        render(<Logo className={customClass} />);

        const logoContainer = screen.getByRole('button').parentElement;
        expect(logoContainer).toHaveClass(customClass);
    });

    it('should have proper accessibility attributes', () => {
        render(<Logo />);

        const logo = screen.getByRole('button');
        expect(logo).toHaveAttribute('type', 'button');
        expect(logo).toHaveAttribute('aria-label', 'Go to home');
    });

    it('should have hover styles', () => {
        render(<Logo />);

        const logo = screen.getByRole('button');
        expect(logo).toHaveClass('hover:text-stone-950', 'transition-colors');
    });

    it('should match snapshot', () => {
        const { container } = render(<Logo />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
