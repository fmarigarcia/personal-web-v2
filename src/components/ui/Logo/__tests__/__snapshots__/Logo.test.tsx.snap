// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot 1`] = `
<div
  class="flex items-center"
>
  <button
    aria-label="Go to home"
    class="text-xl font-bold text-gray-900 hover:text-blue-600 transition-colors"
    type="button"
  >
    FM
  </button>
</div>
`;
